lib/src/htmlfilesetserv/HTMLFileSetHTTPServer.java

```diff
@@ -662,17 +665,19 @@ private AuthToken getToken(final HttpServletRequest request)
 			return auth.validateToken(at);
 		}
 		if (request.getCookies() != null) {
-			for (final Cookie c: request.getCookies()) {
-				if (c.getName().equals(TOKEN_COOKIE_NAME) && !c.getValue().isEmpty()) {
-					return auth.validateToken(unmungeCookiePerShane(c.getValue()));
-				}
-			}
-			// hacky hack hack - remove later when a final solution for cookies is found
-			for (final Cookie c: request.getCookies()) {
-				if (c.getName().equals(TOKEN_COOKIE2_NAME) && !c.getValue().isEmpty()) {
-					return auth.validateToken(unmungeCookiePerShane(c.getValue()));
-				}
-			}
+		    for (String cookieKey : new String[] {TOKEN_COOKIE_NAME, TOKEN_COOKIE2_NAME}) {
+		        for (final Cookie c: request.getCookies()) {
+		            if (c.getName().equals(cookieKey) && !c.getValue().isEmpty()) {
+		                String cookieValue = c.getValue();
+		                String decodedValue = URLDecoder.decode(cookieValue, 
+		                        StandardCharsets.UTF_8.name());
+		                if (decodedValue.contains("\\|") || decodedValue.contains("=")) {
```

Comment:
Theoretically the new tokens are not guaranteed to be any particular string, so this is a little hairy, but since this code will go away as soon as auth2 is complete I don't think it's an issue.